source/common/ssl/context_impl.cc

```diff
@@ -209,7 +209,7 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
         ASN1_STRING* str = altname->d.uniformResourceIdentifier;
         char* crt_san = reinterpret_cast<char*>(ASN1_STRING_data(str));
```

Comment:
It just occurred to me that there is `ASN1_STRING_length()`, which is O(1), so this could be changed to:

    std::string crt_san(reinterpret_cast<char *>(ASN1_STRING_data(str)), ASN1_STRING_length(str));

and then we could compare strings only if `crt_san.length() >= uriPattern.length()` and use `std::string::compare()` again.

Sorry for not catching this yesterday!